common/src/main/java/com/github/khanshoaib3/minecraft_access/features/XPIndicator.java

```diff
@@ -5,37 +5,39 @@
 import lombok.extern.slf4j.Slf4j;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.client.resource.language.I18n;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * This feature speaks when the player xp level is increased or decreased.
  */
 @Slf4j
 public class XPIndicator {
-    private int previousXPLevel;
+    @Nullable
+    private Integer previousXPLevel = null;
 
     public XPIndicator() {
-        previousXPLevel = -999;
     }
 
     public void update() {
-        try {
-            MinecraftClient minecraftClient = MinecraftClient.getInstance();
-            if (minecraftClient == null) return;
-            if (minecraftClient.world == null) return;
-            if (minecraftClient.player == null) return;
-            if (minecraftClient.currentScreen != null) return;
+        MinecraftClient minecraftClient = MinecraftClient.getInstance();
+        if (minecraftClient == null) return;
+        if (minecraftClient.world == null) return;
+        if (minecraftClient.player == null) return;
+        if (minecraftClient.currentScreen != null) return;
 
-            int currentXPLevel = PlayerUtils.getExperienceLevel();
-            if (previousXPLevel == currentXPLevel) return;
-
-            boolean increased = previousXPLevel < currentXPLevel;
+        int currentXPLevel = PlayerUtils.getExperienceLevel();
+        if (previousXPLevel == null) {
             previousXPLevel = currentXPLevel;
-
-            String toSpeak = (increased) ? I18n.translate("minecraft_access.xp_indicator.increased", currentXPLevel)
-                    : I18n.translate("minecraft_access.xp_indicator.decreased", currentXPLevel);
-            MainClass.speakWithNarrator(toSpeak, true);
-        } catch (Exception e) {
-            log.error("An error occurred in xp indicator.", e);
+            return;
+        } else if (previousXPLevel == currentXPLevel) {
```

Comment:
Using an else if here is redundant because there is already a return statement in the first branch.
```suggestion
        }
        if (previousXPLevel == currentXPLevel) {
```